app/src/main/java/org/dhis2/usescases/coodinates/CoordinatesView.java

```diff
@@ -148,6 +148,28 @@ private void setLayout() {
         position.setOnClickListener(this);
         map.setOnClickListener(this);
         clearButton.setOnClickListener(this);
+
+        longitude.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if(!hasFocus)
+                    if (!longitude.getText().toString().isEmpty()) {
+                        Double lon = DoubleExtensionsKt.truncate(getLongitude());
+                        longitude.setText(lon.toString());
```

Comment:
I think you need to have a regex or validate the entry. Otherwise if I don't write the coordinates properly after changing focus this could crash?

What if I write "1." and change focus?

Reply:
In the first two cases a "0" is added appropriately. There wasn't any validation before but will added. The same happens with the "-" sign as the ".".